crates/noirc_evaluator/src/brillig/brillig_gen/brillig_block.rs

```diff
@@ -712,6 +815,25 @@ impl<'block> BrilligBlock<'block> {
         }
     }
 
+    fn update_slice_length(
```

Comment:
We could note here in a comment that the length inside the HeapVector is the number of fields that the vector contains, but the slice length is the user-level length of the slice, corresponding to .len()

Reply:
I added a similar comment that I have in SSA and also the difference between HeapVector and the user-level length of the slice.